copy/copy.go

```diff
@@ -92,6 +93,7 @@ type Options struct {
 	ReportWriter     io.Writer
 	SourceCtx        *types.SystemContext
 	DestinationCtx   *types.SystemContext
+	LayerCopyHook    func(types.BlobInfo) bool
```

Comment:
… and does this _really_ have to be a function? Could this be a set, as per #159 ?

Or, more generally, if we add “delete layer” here, pretty soon we will be adding an “add a layer at this index”. What would _that_ interface look like, and can we define that instead right now? (Yeah, that would be making this PR more complex. Perhaps we should just merge this, or the “layer subset” non-hook variant, right now, with a `// Warning: API likely to change`, and not block on the higher-level discussions?)